popup.js

```diff
@@ -1,38 +1,46 @@
 let active = document.getElementById('active');
+const reloadInput = document.getElementById('reload-int-input');
 
+// Maintain checked input box if this window has active reload
 chrome.tabs.query({ currentWindow: true, active: true }, function (tabs) {
   let tab = tabs[0];
   chrome.storage.local.get(tab.id.toString(), (items) => {
       if(items[tab.id]){
           console.log("found!");
           active.checked = true;
-          
       }
   });
 });
 
+// Save current window tab to Chrome storage
 active.onchange = function(event) {
-  if(event.target.checked){
-    let interval = 10000;
+  if (event.target.checked) {
     
     chrome.tabs.query({ currentWindow: true, active: true }, function (tabs) {
-      console.log(tabs[0]);
       let tab = tabs[0];
-      let tostore = {};
-      tostore[tab.id] = true;
-      chrome.storage.local.set(tostore);
+      let storedObj = {};
+      storedObj[tab.id] = true;
+      chrome.storage.local.set(storedObj);
     });
+
     chrome.browserAction.setBadgeText({text: 'ON'});
-    
-  }
-  else{
+
+  } else {
+
     chrome.tabs.query({ currentWindow: true, active: true }, function (tabs) {
-      console.log(tabs[0]);
       let tab = tabs[0];
-      let tostore = {};
-      tostore[tab.id] = false;
-      chrome.storage.local.set(tostore);
+      let storedObj = {};
+      storedObj[tab.id] = false;
+      chrome.storage.local.set(storedObj);
     });
+
     chrome.browserAction.setBadgeText({text: 'OFF'});
+
   }
+}
+
+// Save reload interval parameter to Chrome storage
+reloadInput.onchange = function(event) {
```

Comment:
Do you think this again could be customisable for each tab?
What you are doing here is setting same input value for each tab! So if someone goes on a tab and sets value as 20 and expects page to be reloaded every 20 seconds and then he sets at 10 seconds for another tab, both the tabs will start refreshing every 10 seconds!

Reply:
To be clear, I was primarily looking for the time interval to be customisable independently for each tab. Sorry for not being explicit in the issue description.

Reply:
I think the functionality is as expected, the refreshes occur at the defined time intervals as we register the setInterval as soon as the store is updated so this way the user defined interval is linked to it. But this does not feel correct, does it? 

I would store the user defined intervals. That would also help me show the reload interval to the user.

Reply:
Done. I changed the value stored in `chrome.storage.local` as an object in which the key equals `tab.id` of the tab in question. Its properties are `{ active: bool, input: num }`. That way, each tab has its own set up parameters. I tested this on 5 different tabs and it works simultaneously with their own interval parameters.